src/main/java/com/shopping_app/shoppingApp/repository/AddressRepository.java

```diff
@@ -3,6 +3,9 @@
 import com.shopping_app.shoppingApp.domain.Address;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public interface AddressRepository extends AbstractRepository<Address> {
```

Comment:
Why do we need `AbstractRepository` here?

Reply:
abstractRepository will extends JPA with primary key in type 'long' which all other class using the same  , to avoid duplication added this class and extending this to other class
 and here addressrepository also uses jpa with primary key as long type so i used here

Reply:
Avoiding duplication is okay, but what benefit does it provide here?

Even for duplication there is nothing much that is duplicated. Now it's importing Abstract repository in all the files instead of JPA repository.

Reply:
removed abstract repository